decode_response.go

```diff
@@ -70,6 +72,13 @@ func (sp *SAMLServiceProvider) ValidateEncodedResponse(encodedResponse string) (
 	doc := etree.NewDocument()
 	err = doc.ReadFromBytes(raw)
 	if err != nil {
+		// Attempt to inflate the response in case it happens to be compressed (as with one case at saml.oktadev.com)
+		buf, flateErr := ioutil.ReadAll(flate.NewReader(bytes.NewReader(raw)))
+		if flateErr == nil {
+			err = doc.ReadFromBytes(buf)
+		}
+	}
+	if err != nil {
 		return nil, err
 	}
```

Comment:
It might be a bit too subtle or clever, but my intent was:
1. Attempt to parse as base64-encoded XML (the common case)
2. If that fails (`err != nil`), then attempt to parse the inflated base64-encoded body as XML
3. If the inflating fails (i.e. it's not compressed), we would return the original `err`
4. If parsing the inflated XML fails, we save the error as `err`; if the parsing succeeds, then `err` is cleared
5. If `err != nil` after any of this, then return early

The main point is that if the SAML response happens to be invalid XML but not compressed, then the inflate call would mask the error.